python/mxnet/kvstore.py

```diff
@@ -35,6 +35,32 @@ def _ctype_key_value(keys, vals):
             c_vals += c_val_i
         return (c_array(ctypes.c_int, c_keys), c_array(NDArrayHandle, c_vals))
 
+def _ctype_str_key_value(keys, vals):
+    names = []
+    if isinstance(keys, str):
+        if isinstance(vals, NDArray):
+            names.append(c_str(keys))
+            return (c_array(ctypes.c_char_p, names),
+                    c_array(NDArrayHandle, [vals.handle]))
+        else:
+            for value in vals:
+                assert(isinstance(value, NDArray))
+            return (c_array(ctypes.c_char_p, [c_str(keys)] * len(vals)),
+                    c_array(NDArrayHandle, [value.handle for value in vals]))
+    else:
+        assert(len(keys) == len(vals))
+        for k in keys:
+            assert(isinstance(k, str))
+        c_keys = []
+        c_vals = []
+        for key, val in zip(keys, vals):
+            c_key_i, c_val_i = _ctype_str_key_value(key, val)
+            c_keys += c_key_i
+            c_vals += c_val_i
+        return (c_array(ctypes.c_char_p, c_keys), c_array(NDArrayHandle, c_vals))
+
+def _use_str_keys(key):
```

Comment:
sure. Do we still want to keep the `MXKVStorePull(..., mx_uint *key)`  API then? Currently I need to create a separate one `MXKVStorePullEx `